### ISSUE TITLE
Add support for defining the CBRAIN type of Boutiques output files

### ISSUE THREAD

**Issue by prioux** [aces/cbrain#1027]
When a task described by a Boutiqes descriptor finishes, it creates as output either a SingleFile or a FileCollection based on whether the output is a normal file or a directory. 

It woudl be nice to support a special extension in the descriptor to let the CBRAIN integrator pick a better CBRAIN type. Something like:

```json
"custom": {
  "cbrain:output_types": {
    "file_out": "fMRIPrepOutput",
    "log_out": "LogFile"
  }
}
```

where the keys in the `cbrain_output_types` object are the IDs of the outputs defined in the descriptor.

**Comment by prioux** [comment]
We could use the very same mechanism to enforce a CBRAIN file type for inputs of type 'File' too!

**Comment by MontrealSergiy** [comment]
are there situation when several output files generated?

**Comment by prioux** [comment]
None come to mind right now, but no harm is being general.

**Comment by prioux** [comment]
I found several tasks with multiple output files.

**Comment by MontrealSergiy** [comment]
Seems resolved by `BoutiquesOutputFileTypeSetter`

**Comment by prioux** [comment]
Implemented by #1199
